src/RulesTest/PropertyConditions/NameTest.cs

```diff
@@ -279,5 +279,21 @@ public void TestTextWithDisallowedPlatformProperties()
                 Assert.IsTrue(Misc.NameRequired.Matches(e));
             } // using
         }
+
+        [TestMethod]
+        public void TestExcludedClassNames()
+        {
+            string[] excludedClassNames = { "Popup", "ContextMenu" };
+
+            using (var e = new MockA11yElement())
+            {
+                e.Framework = FrameworkId.WPF;
```

Comment:
What do you think about adding a test that should assert true before you set the class names? To me, it at least indicates that it's the class names causing the condition not to match.

Reply:
Added both positive and negative tests. I set the ControlType as Window to satisfy the class condition